### PR TITLE
Import the Ops Research tools in TensorFlow as well as their dependencies

### PR DESCRIPTION
**Pull request by benoitsteiner**

### CODE REVIEW

**Comment by gunan** [comment]
Looks like windows build flaked.
jenkins, test this please.

File: tensorflow/workspace.bzl [hunk 1]
```diff
@@ -228,6 +239,29 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
       strip_prefix = "protobuf-2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a",
   )
 
+  # We need to import the protobuf library under the names com_google_protobuf
+  # and com_google_protobuf_cc to enable proto_library support in bazel.
+  # Unfortunately there is no way to alias http_archives at the moment.
+  native.http_archive(
+      name = "com_google_protobuf",
```
Comment (gunan) [inline tensorflow/workspace.bzl]: This is not mentioned in cl description. is it maybe a merge artifact?

File: tensorflow/workspace.bzl [hunk 2]
```diff
@@ -228,6 +239,29 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
...
+      urls = [
+          "http://bazel-mirror.storage.googleapis.com/github.com/google/protobuf/archive/2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a.tar.gz",
+          "https://github.com/google/protobuf/archive/2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a.tar.gz",
+      ],
+      sha256 = "e5d3d4e227a0f7afb8745df049bbd4d55474b158ca5aaa2a0e31099af24be1d0",
+      strip_prefix = "protobuf-2b7430d96aeff2bb624c8d52182ff5e4b9f7f18a",
+  )
+
+  native.http_archive(
+      name = "com_google_protobuf_cc",
```
Comment (gunan) [inline tensorflow/workspace.bzl]: ditto above.

File: tensorflow/workspace.bzl [hunk 3]
```diff
@@ -249,6 +283,12 @@ def tf_workspace(path_prefix = "", tf_repo_name = ""):
       actual = "@gmock_archive//:gtest_main",
   )
 
+  native.git_repository(
```
Comment (gunan) [inline tensorflow/workspace.bzl]: ditto above.

**Review by gunan — CHANGES_REQUESTED** [review]

File: tensorflow/workspace.bzl (hunk 1 above)
Comment (benoitsteiner) [inline tensorflow/workspace.bzl]: It's needed by the ops research tools: they have a protobuf_library bazel rule, which requires the workspace to import the protobuf code under the names com_google_protobuf and com_google_protobuf_cc.

File: tensorflow/workspace.bzl (hunk 3 above)
Comment (benoitsteiner) [inline tensorflow/workspace.bzl]: This is also needed by the ops research tools, that depend on the gflags library.

**Comment by gunan** [comment]
I will approve this now to unblock work.
But we should explore how we can dedupe proto libraries @jart for help on that
@vrv about gflags addition. It is a dependency of a dependency, but I think we would like to avoid using it, right?

**Review by gunan — APPROVED** [review]

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by benoitsteiner** [comment]
If needs be, we can always ask the ops research guys to use another library to manage their flags.

**Pull request merged**